File: splitxix33/Splitxix33.java
// splitxix33: a splitmix64 with memorable constants
// This is free and unencumbered software released into the public domain.
public class Splitxix33 {
    public long s;
    public Splitxix33 (long seed) { s = seed; }

    public long next() {
        long r = (this.s += 1111111111111111111L);
        r ^= r >>> 33; r *= 1111111111111111111L;
        r ^= r >>> 33; r *= 1111111111111111111L;
        r ^= r >>> 33;
        return r;
    }

    public static void main(String[] args) {
        Splitxix33 g[] = {
            new Splitxix33(0), new Splitxix33(1),
            new Splitxix33(2), new Splitxix33(3),
        };
        for (int i = 0; i < 40; i++) {
            System.out.printf("%016x %016x %016x %016x\n",
                              g[0].next(), g[1].next(),
                              g[2].next(), g[3].next());
        }
    }
}

File: splitxix33/splitxix33.py
# splitxix33: a splitmix64 with memorable constants
# This is free and unencumbered software released into the public domain.
import numpy
import typing

numpy.seterr(over="ignore")

def splitxix33(seed: int) -> typing.Generator[int, None, None]:
    m: numpy.uint64 = numpy.uint64(1111111111111111111)
    s: numpy.uint64 = numpy.uint64(33)
    x: numpy.uint64 = numpy.uint64(seed)
    r: numpy.uint64
    while True:
        x += m
        r  = x; r ^= r >> s
        r *= m; r ^= r >> s
        r *= m; r ^= r >> s
        yield int(r)


# Example
g = tuple(splitxix33(i) for i in range(4))
for _ in range(40):
    print(*(f"{next(g):016x}" for g in g))
